### DISCUSSION TITLE
Check if battery is less than %

### DISCUSSION THREAD

**Discussion by unknownsolo** [gopro/labs#692] [Q&A]
This is mainly for dash cam mode. Since the battery doesn't charge while on USB power and recording, the battery depletes over time. Is there any way to set up an if statement with something like if battery is less than 70%, charge up to 80% then record? Or even better, charge while recording, but only up to 80%, so it's not constantly charging. 

I know this may delay recording in some instances, but at least it will keep the battery charged instead of manually charging it from time to time.

**Comment by unknownsolo** [comment]
Also, to clarify, I could remove the battery altogether, but then when USB power is switched off, the recording would just cut off instead of safely saving the file. Would that cause any issues? If not, I would remove the battery and call it a day.

**Reply by dnewman-gpsw** [reply]
File is automatically repaired on next boot. On HERO11 and 12 it is straight forward to test battery level script conditions based on that. Use https://gopro.github.io/labs/build/

**Reply by unknownsolo** [reply]
Thank you, I didn't notice the battery sensor, does this look right?


![Screenshot 2023-12-30 at 11 22 46 PM](https://github.com/gopro/labs/assets/86228066/4149af7c-4109-4779-82ea-4ad948940bec)

![New Project-12_30_2023, 23_22_08](https://github.com/gopro/labs/assets/86228066/cb17805d-05cc-44ed-b563-a56c1f65564e)

**Reply by dnewman-gpsw** [reply]
Looks good
